ui/apps/platform/src/Components/CompoundSearchFilter/utils/utils.tsx

```diff
@@ -19,8 +19,16 @@ export const conditionMap = {
     'Is less than': '<',
 };
 
+export const dateConditionMap = {
+    Before: '<',
+    On: '', // "=" doesn't work but we can omit the condition to work like an equals
+    After: '>',
+};
+
 export const conditions = Object.keys(conditionMap);
 
+export const dateConditions = Object.keys(dateConditionMap);
+
```

Comment:
With an `as const` on `dateConditionMap` and `conditionMap` above, I think this is one of the few places it is appropriate to use a type assertion on the derived values:
```suggestion
export const dateConditionMap = {
    Before: '<',
    On: '', // "=" doesn't work but we can omit the condition to work like an equals
    After: '>',
} as const;

export const conditions = Object.keys(conditionMap) as unknown as keyof typeof conditionMap;

export const dateConditions = Object.keys(dateConditionMap) as unknown as keyof typeof dateConditionMap;
```

Although this comes up semi-frequently enough, so I bet at some point we could create a global util that correctly extracts the keys from a `readonly` constant object.

Reply:
Sounds good. I changed it to `as unknown as keyof (typeof dateConditionMap)[]` because it should be an array.